### DISCUSSION TITLE
4.3.1

### DISCUSSION THREAD

**Discussion by jawngee** [Interfacelab/ilab-media-tools#173] [General]
* **NOW REQUIRES PHP 7.4**  Installing on PHP < 7.4 will not work and result in errors.
* Sign up to be notified about our new product for WordPress coming in April 2022: [Preflight for WordPress](https://preflight.ju.mp)
* Fixed compatibility with BuddyPress 6.x
* Fixed compatibility with BuddyBoss 1.8.x including video.  **Note:** Mux encoding does not work with BuddyBoss and it's impossible to make it work.  You can have it enabled and Mux will encode videos but the videos that are played on the front end will be the uploaded MP4 source.  It's best to turn Mux off if you are using it with BuddyBoss.
* Added new Wasabi and S3 regions.
* All third party libraries have been updated to the latest versions.
* Fixed MUX gutenberg block registration
* Filmstrip generation with Mux now warns you if GD is not installed
* Media Cloud now makes **EWWW Image Optimizer** run during the upload process which fixes a lot of issues with other plugins like Elementor.  You can disable this in **Cloud Storage Settings** if your uploads have become unbearably slow.
* Fixed compatibility with **EWWW Image Optimizer** bulk optimizer.
* Generated `.webp` file names are stored in S3 metadata.
* `.webp` files are deleted from cloud storage when deleting an upload.
* When queueing deletes, you can now specify the delay in minutes before items in the queue get processed.
* Built-in image optimizer now properly queues deletes if that setting is enabled.
* For the free version, if you are using Elementor with an image optimizer that isn't **EWWW Image Optimizer**, you are going to have problems and that configuration isn't supported.  You should consider a switch to EWWW or upgrading to the premium version.
* For the premium version, if you are using Elementor with an image optimizer that isn't **EWWW Image Optimizer**, make sure to turn on **Queue Deletes** in **Cloud Storage Settings** and **Auto Update Elementor** in **Integration Settings**.
* Fix for images specified in the customizer when using the Astra theme.
* PDF thumbnails generated with the ImageMagick extension are now uploaded to cloud storage properly.
* Fixed an error with PDF uploads and ImageMagick that would prevent the PDF from being uploaded to cloud storage in certain circumstances.
* Fixed the Render PDF functionality of imgix to now properly render PDFs.
* Large PDF uploads would cause a fatal memory error on some systems.  Now, you can now upload PDFs of any size without issues (subject to WordPress and web server limitations).
* PDF upload speed should be greatly improved.  If you don't care about generating PDF preview images, you can improve it even more by turning off **Extract PDF Page Size** in the **Image and PDF Upload Handling** section of **Cloud Storage Settings**.
* Added a new setting **Background Only When Using the Media Library** that, when enabled, limits image optimizations to only run in the background when using the WordPress Media Library pages in the admin, otherwise the image optimizations will run during the upload.  Enabling this could improve compatibility with some plugins.
* Image Optimizations that fail would prevent the upload from being transferred to cloud storage.  That has been fixed.
* Updated the Imagify Image Optimization driver.
* Background optimizations now start a lot a sooner.
* Added **Update URLs** task to search and replace URLs in the WordPress database.
* Added a switch to the **Migrate to Cloud** task that will do a search and replace for migrated URLs during the migration process.
* When files are deleted from the server, everything about the deletion is now logged when debugging is enabled.
* Fixed issues with WordPress 5.8 and 5.9
* Fixed compatibility with PHP 7.4 and 8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Interfacelab/ilab-media-tools/releases/tag/'>4.3.1</a>.</em>

**Comment by studiomonkriss** [comment]
I've got the latest version running and im using OWWW. I can see the webp versions of the images are being saved onto S3... but they do not appear to be being served on the website. I've turned on the webp delivery methods on EWWW but still nothing. Do you have any ideas?
